arshal.go

```diff
@@ -544,6 +511,9 @@ func newAddressableValue(t reflect.Type) addressableValue {
 	return addressableValue{reflect.New(t).Elem(), true}
 }
 
+// TODO: Remove *jsonopts.Struct argument from [marshaler] and [unmarshaler].
```

Comment:
I suspect this will speed up the performance of marshaling and unmarshaling since each recursive call doesn't need to keep passing the same *Struct pointer on the stack.